platform/fabric/core/generic/committer/endorsertx.go

```diff
@@ -18,83 +18,107 @@ import (
 type ValidationFlags []uint8
 
 func (c *committer) handleEndorserTransaction(block *common.Block, i int, event *TxEvent, env *common.Envelope, chHdr *common.ChannelHeader) {
+	txID := chHdr.TxId
+	event.Txid = txID
+
+	validationCode := pb.TxValidationCode(ValidationFlags(block.Metadata.Metadata[common.BlockMetadataIndex_TRANSACTIONS_FILTER])[i])
+	switch validationCode {
+	case pb.TxValidationCode_VALID:
+		if err := c.CommitEndorserTransaction(txID, block, i, env, event); err != nil {
+			logger.Panicf("failed committing transaction [%s] with err [%s]", txID, err)
+		}
+	default:
+		if err := c.DiscardEndorserTransaction(txID, block, event, validationCode); err != nil {
+			logger.Panicf("failed discarding transaction [%s] with err [%s]", txID, err)
+		}
+	}
+}
+
+// CommitEndorserTransaction commits the transaction to the vault
+func (c *committer) CommitEndorserTransaction(txID string, block *common.Block, indexInBlock int, env *common.Envelope, event *TxEvent) error {
 	committer, err := c.network.Committer(c.channel)
 	if err != nil {
 		logger.Panicf("Cannot get Committer [%s]", err)
```

Comment:
I suggest to just return the error here since you are checking for it in `handleEndorserTransaction` anyway

Reply:
If you use `logger.Panic` inside a method, you should document its behaviour. Better just return the error and panic elsewhere if really needed.